### PR TITLE
SWC-6292

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

**Comment by jay-hodgson** [comment]
Wiki IDU Report widget now uses the SRC component:
Screen Shot 2022-11-02 at 2 30 13 PM

There's also this old Generate IDU Report button in the AR UI:
Screen Shot 2022-11-02 at 2 28 47 PM

I now utilize the IDU widget.  So it will not generate the formatted markdown now, but it will ask ACT to use the dynamic wiki widget instead:
Screen Shot 2022-11-02 at 2 42 48 PM

File: src/main/java/org/sagebionetworks/web/client/PortalGinInjector.java [hunk 1]
```diff
@@ -56,7 +56,6 @@
 import org.sagebionetworks.web.client.widget.RadioWidget;
 import org.sagebionetworks.web.client.widget.accessrequirements.ACTAccessRequirementWidget;
 import org.sagebionetworks.web.client.widget.accessrequirements.AccessRequirementWidget;
-import org.sagebionetworks.web.client.widget.accessrequirements.IntendedDataUseGenerator;
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/PortalGinInjector.java:59]: Deleted IntendedDataUseGenerator

File: src/main/java/org/sagebionetworks/web/client/PortalGinModule.java [hunk 2]
```diff
@@ -1455,7 +1459,8 @@ protected void configure() {
         bind(SynapseContextPropsProvider.class).to(SynapseContextPropsProviderImpl.class);
         bind(AddToDownloadListV2.class).to(AddToDownloadListV2Impl.class);
         bind(OpenDataView.class).to(OpenDataViewImpl.class);
-
         bind(QueryClientProvider.class).to(QueryClientProviderImpl.class).in(Singleton.class);
+        bind(IntendedDataUseReportWidgetView.class).to(IntendedDataUseReportWidgetViewImpl.class);
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/PortalGinModule.java:1463]: Added an IntendedDataUseReport view implementation (that utilizes the SRC component)

File: src/main/java/org/sagebionetworks/web/client/PortalGinModule.java [hunk 3]
```diff
@@ -1455,7 +1459,8 @@ protected void configure() {
         bind(SynapseContextPropsProvider.class).to(SynapseContextPropsProviderImpl.class);
         bind(AddToDownloadListV2.class).to(AddToDownloadListV2Impl.class);
         bind(OpenDataView.class).to(OpenDataViewImpl.class);
-
         bind(QueryClientProvider.class).to(QueryClientProviderImpl.class).in(Singleton.class);
+        bind(IntendedDataUseReportWidgetView.class).to(IntendedDataUseReportWidgetViewImpl.class);
+        bind(DialogView.class).to(Dialog.class);
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/PortalGinModule.java:1464]: Added a general DialogView that can be used in a Presenter to build up the UI programmatically.  In this case, wrapping the IDUReport widget in a modal.

File: src/main/java/org/sagebionetworks/web/client/widget/accessrequirements/IntendedDataUseReportButton.java [hunk 4]
```diff
@@ -18,33 +22,31 @@ public class IntendedDataUseReportButton implements IsWidget {
...
 		});
 	}
 
-	public void showIDUs(String md) {
-		copyTextModal.configure(IDU_MODAL_TITLE, IDU_MODAL_FIELD_NAME, md.toString());
-		copyTextModal.show();
-	}
-
 	public void configure(AccessRequirement ar) {
+		synAlert.showError("Please use the dynamic IDU Report wiki widget instead of this snapshot. ${iduReport?accessRestrictionId="+ar.getId()+"}");
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/accessrequirements/IntendedDataUseReportButton.java:49]: ACT only UI.  If an ACT member ever clicks this, point them to the wiki widget.  When the ACT Dashboard is released, my hope is that we can remove the IntendedDataUseReportButton altogether

Comment (nickgros) [reply src/main/java/org/sagebionetworks/web/client/widget/accessrequirements/IntendedDataUseReportButton.java:49]: I don't think we have a place for that anywhere in the new dashboard (yet). Clicking an AR ID takes you to the old AR page.

**Review by nickgros — APPROVED** [review]

**Pull request merged**